docs/NEW_RULES.md

```diff
@@ -1,42 +1,240 @@
+      .setFeedEndDate(feedEndDate)
+      .build();
+}
+```
+
+2️⃣ Create a [`NoticeContainer`](../core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java):
+```
+NoticeContainer container = new NoticeContainer();
+```
+
+3️⃣ Execute the validator one the previously defined parameters (`GtfsEntity` and `NoticeContainer`).
+```
+validateFeedInfo(createFeedInfo(GtfsDate.fromLocalDate(TEST_NOW.toLocalDate().plusDays(7))))
+```
+4️⃣ Verify the content of `NoticeContainer`: 
+```
+assertThat(
+        validateFeedInfo(createFeedInfo(GtfsDate.fromLocalDate(TEST_NOW.toLocalDate().plusDays(7)))))
+    .containsExactly(
+        new FeedExpirationDateNotice(
+            1,
+            GtfsDate.fromLocalDate(TEST_NOW.toLocalDate()),
+            GtfsDate.fromLocalDate(TEST_NOW.toLocalDate().plusDays(7)),
+            GtfsDate.fromLocalDate(TEST_NOW.toLocalDate().plusDays(30))));
+```
+
+One can also refer to [`this example`](../main/src/test/java/org/mobilitydata/gtfsvalidator/validator/FeedExpirationDateValidatorTest.java).
+ 
 ### Test a `FileValidator`
-1️⃣ Create a class constant `NoticeContainer` and annotate it `@Mock`
-
-2️⃣ Declare the validator to be tested as a class constat and annotate it `@InjectMocks`: mockito will inject the `NoticeContainer`.
-
-3️⃣ Implement `initMocks`  method. Annotate it `@Before` so that it is ran before all tests of the class.
-
-4️⃣ Stub all required methods for all mocks used.
-
-5️⃣ Verify interactions with the `NoticeContainer`
-
-6️⃣ Verify interactions with all mocks
-
-## Resources
-
-[Use Mockito to Mock Autowired Fields](https://dzone.com/articles/use-mockito-mock-autowired)
+1️⃣ Create an instance of the validator to test
+```
+    TripUsabilityValidator tripUsabilityValidator = new TripUsabilityValidator();
+```
+
+2️⃣ Create the relevant [`GtfsTableContainers`](../core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsTableContainer.java) and inject them in the validator
+```
+    tripUsabilityValidator.tripTable =
+        createTripTable(
+            noticeContainer,
+            ImmutableList.of(
+                createTrip(1, "route id value", "service id value", "t0"),
+                createTrip(3, "route id value", "service id value", "t1")));
+    tripUsabilityValidator.stopTimeTable =
+        createStopTimeTable(
+            noticeContainer,
+            ImmutableList.of(
+                createStopTime(0, "t0", "s0", 2),
+                createStopTime(2, "t0", "s1", 3),
+                createStopTime(0, "t1", "s3", 5),
+                createStopTime(2, "t1", "s4", 9)));
+```
+
+3️⃣ Execute the validator `.validate()` method
+```
+    underTest.validate(noticeContainer);
+```
+
+4️⃣ Verify the content of `NoticeContainer`.
+```
+    assertThat(noticeContainer.getValidationNotices()).isEmpty();
+```
+
+
+One can also refer to [`this example`](../main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TripUsabilityValidatorTest.java).
```

Comment:
I would put the example link before the steps, that way the reader can click straight to that if they want.